goldie.go

```diff
@@ -133,21 +133,17 @@ func compare(name string, data interface{}, actualData []byte) error {
 }
 
 // ensureDir will create the fixture folder if it does not already exist.
-func ensureDir(loc string) error {
+func ensureDir(loc string) (err error) {
```

Comment:
I like the fact that the function gets more concise, but I think we should explicitly return in the cases, to ensure that the code stops executing there.

It'd think that this version of your changes are more readable.

```
func ensureDir(loc string) error {
	s, err := os.Stat(loc)
	switch {
	case err != nil && os.IsNotExist(err):
		// the location does not exist, so make directories to there
		return os.MkdirAll(loc, DirPerms)
	case err == nil && !s.IsDir():
		return newErrFixtureDirectoryIsFile(loc)
	}

	return nil
}
```

What do you think?

Reply:
Subjectively, I would prefer a single return, but can be and so. In the example above, not all error cases returned by are handled  `os.Stat(loc)` and need return err instead of `nil`.

Reply:
Fair enough 👍 , I'll merge this!